src/include/cache_inode.h

```diff
@@ -969,6 +973,32 @@ cache_inode_refresh_attrs(cache_entry_t *entry)
 		entry->obj_handle->attrs->acl = NULL;
 	}
 
+	if ((entry->type == DIRECTORY) &&
```

Comment:
This can be done in trust_attrs function (like in the original code). If attrs are refreshed through some other path then cache_inode_is_attrs_valid may always return true not giving a chance to invalidate dentries. This should be taken care of.